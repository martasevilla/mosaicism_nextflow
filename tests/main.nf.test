nextflow_pipeline {

    name "Test Workflow main.nf"
    script "main.nf"

    test("Minimal test") {

        when {
            params {
                input  = "https://raw.githubusercontent.com/CIBERER/GdTBioinfo-nf-mosaicism/dev/testdata/samplesheet_test.csv"
                fasta = "https://zenodo.org/records/11185276/files/chr17_var_regions.fa"
                fai = "https://zenodo.org/records/11185276/files/chr17_var_regions.fa.fai"
                chrom_sizes = "https://zenodo.org/records/11185276/files/chr17_chrom_sizes.txt"
                germline_resource = "https://zenodo.org/records/11185276/files/chr17_af_gnomad.vcf.gz"
                germline_resource_tbi = "https://zenodo.org/records/11185276/files/chr17_af_gnomad.vcf.gz.tbi"
            }
        }

        then {
            assert workflow.success
            assert snapshot(workflow).match() 
        }

    }

}
