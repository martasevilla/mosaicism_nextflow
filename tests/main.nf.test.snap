{
    "Minimal test": {
        "content": [
            {
                "stderr": [
                    
                ],
                "errorReport": "",
                "exitStatus": 0,
                "failed": false,
                "stdout": [
                    "",
                    ""
                ],
                "errorMessage": "",
                "trace": {
                    "tasksFailed": 0,
                    "tasksCount": 22,
                    "tasksSucceeded": 22
                },
                "name": "workflow",
                "success": true
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.3"
        },
        "timestamp": "2024-07-26T10:36:54.833678505"
    }
}